### ISSUE TITLE
Cache size doesn't update after some key expired?

### ISSUE THREAD

**Issue by ihaolin** [google/guava#2598]
Hello，I write a test code：

    cache<String, Integer> cache =
                CacheBuilder.newBuilder()
                        .expireAfterAccess(10, TimeUnit.SECONDS).build();
    cache.put("test", 1);

    Thread.sleep(5000);
    // output 1:1
    System.out.println(cache.size() + ": " + cache.getIfPresent("test"));

    Thread.sleep(10000);
    // output 1:null
    System.out.println(cache.size() + ": " + cache.getIfPresent("test"));

The cache doesn't update its size after the key expired, it's a bug or others?

Thanks & Regards.

**Comment by lowasser** [comment]
This is specified in so many words in the CacheBuilder Javadoc: "If expireAfterWrite or expireAfterAccess is requested entries may be evicted on each cache modification, on occasional cache accesses, or on calls to Cache.cleanUp(). **Expired entries may be counted by Cache.size(), but will never be visible to read or write operations.**"

Cache.size() also documents that it is only approximate.

**Comment by ben-manes** [comment]
This feature, of using an optional scheduling thread, is released in Caffeine 2.8.

```java
Caffeine.newBuilder()
    .scheduler(Scheduler.systemScheduler()) // Java 9+
    .expireAfterWrite(10, TimeUnit.SECONDS)
    .removalListener(...)
    .build();
```

If you are not on Java 9+ then you will have to supply your own thread, e.g. using a `Scheduler.forScheduledExecutorService(se)`.
